core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -117,4 +124,24 @@ void handleFavicon(HttpRequest request, HttpChannel channel) {
             channel.sendResponse(new BytesRestResponse(FORBIDDEN));
         }
     }
+
+    private static final class ResourceHandlingHttpChannel extends HttpChannel {
+        private final HttpChannel delegate;
+        private final HttpServerTransport transport;
+
+        public ResourceHandlingHttpChannel(HttpChannel delegate, HttpServerTransport transport) {
+            super(delegate.request(), delegate.detailedErrorsEnabled());
```

Comment:
I'll do this but due to package dependencies I'd create the base class in the `rest` package. We sometimes override `RestChannel` in tests and I think it makes sense to keep this. However, I will get rid of `HttpChannel` and related overhead classes then.